src/main/java/software/amazon/cloudwatchlogs/emf/sinks/TCPClient.java

```diff
@@ -65,13 +63,22 @@ public synchronized void sendMessage(String message) {
         }
 
         try {
+            // Write a space to the socket to verify connection before sending event
+            os.write(32);
```

Comment:
Can we write a null character instead?

Reply:
The CloudWatch agent trims all messages before processing them. I'm not sure what it would do with a null character.

Reply:
For reference: https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/parsers/emf/parser.go#L39